hayeon/programmers-0/그림-확대.js

```diff
@@ -0,0 +1,25 @@
+function solution(picture, k) {
+  const answer = [];
+
+  picture.map((el) => {
+    const stretchedFile = el
+      .split("")
+      .map((v) => v.repeat(k))
+      .join("");
+
+    for (let i = 0; i < k; i++) answer.push(stretchedFile);
+  });
+
+  return answer;
+}
```

Comment:
```suggestion
function solution(picture, k) {
  const answer = [];

  picture.map((el) => {
    const stretchedFile = el
      .split("")
      .map((v) => v.repeat(k))
      .join("");

    for (let i = 0; i < k; i++) {
      answer.push(stretchedFile);
    }
  });

  return answer;
}
```
- 풀이잘봤습니다 🙇‍♂️
- for문에서 `절대절대 한 줄 표현식 쓰지마세요` 자스 컨벤션에 맞지 않습니다 꼭 중괄호 남겨주세요. 한 줄 표현식은 if문에서만 해당되는 부분입니다.
- 메서드 체이닝으로 깔끔해 보이는데 생각보다 연산이 많은 것 같습니다 아래 reduce를 활용하면 효율성도 가져갈 수 있을 것 같습니다!
```js
function solution(picture, k) {
  return picture.reduce((acc, cur) => {
    let tempStr = "";

    for (const char of cur) {
      tempStr += char.repeat(k);
    }

    for (let i = 0; i < k; i++) {
      acc.push(tempStr);
    }

    return acc;
  }, []);
}
```